fedimint-server/src/consensus/server.rs

```diff
@@ -283,7 +280,7 @@ async fn confirm_consensus_config_hash(
 
 async fn relay_messages(
     task_group: &mut TaskGroup,
-    mut connections: PeerConnections<Message>,
+    mut connections: ReconnectPeerConnections<Message>,
```

Comment:
I do not understand the question. I used ReconnectPeerConnections here directly because PeerConnections wrapper seemed unnecessary and I did not want to change the IPeerConnections trait to use a sync send method as it was caused a test to fail.

Reply:
In general these `Box<dyn ITrait>` "wrappers" exist where we a) want to be able to plug in different implementations or b) want to be able to mock out implementations for testing. I may have overdone it a bit in the network stack and we still have enough of them left I think, I just wanted to make sure you had sufficient context of why it exists before removing.